download.js

```diff
@@ -40,27 +40,37 @@ function downloadPrebuild (opts, cb) {
           return unpack()
         }
 
-        log.http('request', 'GET ' + downloadUrl)
-        var req = get(downloadUrl, function (err, res) {
-          if (err) return onerror(err)
-          log.http(res.statusCode, downloadUrl)
-          if (res.statusCode !== 200) return onerror()
-          fs.mkdir(util.prebuildCache(), function () {
-            log.info('downloading to @', tempFile)
-            pump(res, fs.createWriteStream(tempFile), function (err) {
-              if (err) return onerror(err)
-              fs.rename(tempFile, cachedPrebuild, function (err) {
-                if (err) return cb(err)
-                log.info('renaming to @', cachedPrebuild)
-                unpack()
+        fs.mkdir(util.prebuildCache(), function () {
+
+          var stream = fs.createWriteStream(tempFile);
+          stream.on('open', function() {
+            request({
+              url: downloadUrl,
+              encoding: null,
+              timeout: 3 * 1000
+            }).on('error', function (err) {
+              return onerror(err)
+            }).on('response', function (res) {
+              log.http(res.statusCode, downloadUrl)
+              if (res.statusCode !== 200) return onerror()
+              log.info('downloading to @', tempFile)
+
+              stream.on('close', function () {
+                fs.rename(tempFile, cachedPrebuild, function (err) {
+                  if (err) return cb(err)
+                  log.info('renaming to @', cachedPrebuild)
+                  unpack()
+                })
               })
-            })
+            }).pipe(stream)
+          })
+          
+          stream.on('error', function(err) {
```

Comment:
stream.on('open') before making the request, because fs.createWriteStream is not immediately available. http://stackoverflow.com/questions/12906694/fs-createwritestream-does-not-immediately-create-file